src/napari_threedee/visualization/lighting_control.py

```diff
@@ -31,6 +34,58 @@ def _on_enable(self):
     def _on_disable(self):
         self._disconnect_events()
 
+    @property
+    def ambient_occlusion(self) -> bool:
+        return self._ambient_occlusion
+
+    @ambient_occlusion.setter
+    def ambient_occlusion(self, value: bool) -> None:
+        if value == self.ambient_occlusion:
+            # if the value isn't changed, do nothing
+            return
+        self._ambient_occlusion = value
+
+        if value is True:
+            self._activate_ambient_occlusion()
+        else:
+            self._deactivate_ambient_occlusion()
+
+    def _activate_ambient_occlusion(self):
+        self._compute_ambient_occlusion()
+
+    def _compute_ambient_occlusion(self):
+        self._pre_ambient_occlusion_values = []
+        for layer, visual in zip(self._selected_layers, self._selected_layer_visuals):
+            vertices, faces, vertex_values = layer.data
+            self._pre_ambient_occlusion_values.append(vertex_values)
+            vertex_normals = igl.per_vertex_normals(vertices, faces)
+            ao = igl.ambient_occlusion(vertices, faces, vertices, vertex_normals, 20)
+            attenuation_factor = 1 - ao
+
+            #
+            attenuated_values = vertex_values * attenuation_factor
+
+            # set the data
+            # meshdata = visual.node._meshdata
+            # meshdata.set_vertex_values(attenuated_values)
+            # visual.node.set_data(meshdata=meshdata)
+
+    def _calc(self, vertices, faces, vertex_values):
+        vertex_normals = igl.per_vertex_normals(vertices, faces)
+        ao = igl.ambient_occlusion(vertices, faces, vertices, vertex_normals, 20)
+        attenuation_factor = 1 - ao
```

Comment:
formatting here is funky

Reply:
I'll clean this up. I need to do another pass to add some events to sync when the data gets updated.